core/presentation/src/main/java/com/nibalk/tasky/core/presentation/utils/BitmapiExt.kt

```diff
@@ -0,0 +1,73 @@
+package com.nibalk.tasky.core.presentation.utils
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.graphics.drawable.Drawable
+import android.net.Uri
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.withContext
+import java.io.ByteArrayOutputStream
+import java.net.URL
+
+suspend fun URL.getCompressedByteArray(): ByteArray =
+    coroutineScope {
+        val getBytes = async(Dispatchers.IO) {
+            val url = this@getCompressedByteArray
+            url.readBytes()
+        }
+        getBytes.await()
+    }
+
+suspend fun Uri.getCompressedByteArray(
+    context: Context,
+    uploadTThreshold: Int = 1024 * 1024 // 1MB
+): ByteArray? = withContext(Dispatchers.IO) {
+    val uri = this@getCompressedByteArray
+    val options = BitmapFactory.Options().apply {
+        inJustDecodeBounds = true // Decode bounds without loading the entire image
+    }
+    val imageSize = options.outWidth * options.outHeight // Approximate image size
+
+    options.inJustDecodeBounds = false
+    val bitmap = BitmapFactory.decodeStream(
+        context.contentResolver.openInputStream(uri),
+        null, options
+    )
+
+    if (imageSize > uploadTThreshold) {
+        bitmap?.toCompressedByteArray()
+    } else {
+        bitmap?.toByteArray()
+    }
```

Comment:
To support cancellation of this coroutine, ideally call `ensureActive()` in between blocking calls. Otherwise, if you cancel the coroutine this function is running in, it will still fully run through, since it doesn't contain any suspending calls that would check for cancellation.

So after `BitmapFactory.decodeStream()`, putting `ensureActive()` will check if the coroutine was cancelled and if so, it won't go into the byte array compression